src/groupNoble.ts

```diff
@@ -0,0 +1,337 @@
+            ...g.params.hashToScalar,
+            expand: 'xmd',
+            DST: decodeDST(dst),
+            m: 1
+        })
+        return new ScalarNb(g, k)
+    }
+}
+
+class EltNb implements Elt {
+    private constructor(public readonly g: GroupNb, private readonly p: ProjPointType<bigint>) {}
+
+    static new(g: GroupNb): EltNb {
+        return new EltNb(g, g.params.Point.ZERO)
+    }
+
+    static gen(g: GroupNb): EltNb {
+        return new EltNb(g, g.params.Point.BASE)
+    }
+
+    isIdentity(): boolean {
+        return this.p.equals(this.g.params.Point.ZERO)
+    }
+
+    isEqual(a: EltNb): boolean {
+        compat(this, a)
+        return this.p.equals(a.p)
+    }
+
+    neg(): EltNb {
+        return new EltNb(this.g, this.p.negate())
+    }
+
+    add(a: EltNb): EltNb {
+        compat(this, a)
+        return new EltNb(this.g, this.p.add(a.p))
+    }
+
+    mul(s: ScalarNb): EltNb {
+        compat(this, s)
+        return new EltNb(this.g, this.p.multiply(s.k))
+    }
+
+    mul2(k1: ScalarNb, a: EltNb, k2: ScalarNb): EltNb {
+        compat(this, k1)
+        compat(this, k2)
+        compat(this, a)
+        const el = this.p.multiplyAndAddUnsafe(a.p, k1.k, k2.k)
+        if (!el) throw new Error('result is zero')
+        return new EltNb(this.g, el)
+    }
+
+    serialize(compressed = true): Uint8Array {
+        if (this.isIdentity()) {
+            return Uint8Array.from([0])
+        }
+        return this.p.toRawBytes(compressed)
+    }
+
+    // size returns the number of bytes of a non-zero element in compressed or uncompressed form.
+    static size(g: GroupNb, compressed = true): number {
+        return 1 + (compressed ? g.size : g.size * 2)
+    }
+
+    private static deser(g: GroupNb, bytes: Uint8Array): EltNb {
+        const point = g.params.Point.fromHex(bytes)
+        point.assertValidity()
+        return new EltNb(g, point)
+    }
+
+    // Deserializes an element, handles both compressed and uncompressed forms.
+    static deserialize(g: GroupNb, bytes: Uint8Array): EltNb {
+        const len = bytes.length
+        switch (true) {
+            case len === 1 && bytes[0] === 0x00:
+                return g.identity()
+            case len === 1 + g.size && (bytes[0] === 0x02 || bytes[0] === 0x03):
+                return EltNb.deser(g, bytes)
+            case len === 1 + 2 * g.size && bytes[0] === 0x04:
+                return EltNb.deser(g, bytes)
+            default:
+                throw errDeserialization(EltNb)
+        }
+    }
+
+    static hash(g: GroupNb, msg: Uint8Array, dst: Uint8Array): EltNb {
+        const h2c = g.params.h2c
+        const p = h2c(msg, { DST: decodeDST(dst) }) as ProjPointType<bigint>
+        return new EltNb(g, p)
+    }
+}
+
+class GroupNb implements Group {
+    static readonly Elt: EltCons = EltNb
+    static readonly Scalar: ScalarCons = ScalarNb
+    static readonly ID = GroupIDs
+    static for(gid: GroupID) {
```

Comment:
Yeah, it's potentially a YAGNI, but it was looking forward to when ristretto/decaf are implemented and it would likely be cleaner to use separate classes. Using new() would mean having to build a single class for all. 

@armfazh 
Does that make sense?

Reply:
not so clear right now, until we have the other groups.
One concern is more about using `for`, can we replace it with `create`, `createFromID` or `fromID`?

Reply:
Ok, will use fromID